### PR TITLE
Wagtail 60 cherry pick token urls

### PR DESCRIPTION
**Pull request by nickmoreton**
This pr is based of of: https://github.com/torchbox-forks/wagtail-sharing/tree/support/wagtail-60

Before it's merged I'll rebase it on the main branch so it's kept inline with main development.

It adds support for the token-urls by adding the changes from this previously cherry picked source: https://github.com/torchbox-forks/wagtail-sharing/tree/cherry-pick-token-urls

It's intended this will be merged to a separate branch and a tag created from it later on.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -37,6 +37,19 @@ jobs:
         python: ['3.8', '3.11']
         django: ['3.2', '4.2']
         wagtail: ['5.1', '5.2']
+        include:
+          - python: '3.12'
+            django: '4.2'
+            wagtail: '5.2'
+          - python: '3.12'
+            django: '4.2'
+            wagtail: '6.0'
```
Comment (katdom13) [inline .github/workflows/test.yml:46]: These should be removed.

File: pyproject.toml [hunk 2]
```diff
@@ -9,14 +9,17 @@ authors = [
     {name = "CFPB", email = "tech@cfpb.gov" }
 ]
 dependencies = [
-    "wagtail>=5.1",
+    "wagtail>=3",
```
Comment (katdom13) [inline pyproject.toml:12]:
```suggestion
    "wagtail>=5.2",
```

**Review by katdom13 — CHANGES_REQUESTED** [review]
Hi @nickmoreton ,
I have comments similar to the ones in the previous PR:
https://github.com/torchbox-forks/wagtail-sharing/pull/11

File: .github/workflows/test.yml (hunk 1 above)
Comment (nickmoreton) [reply .github/workflows/test.yml]: This is now updated after a rebase

File: pyproject.toml (hunk 2 above)
Comment (nickmoreton) [reply pyproject.toml]: This is fixed here: https://github.com/torchbox-forks/wagtail-sharing/pull/12/commits/7f9718562e4fcd5b11d91a9908641395908c8ebe

**Comment by nickmoreton** [comment]
Hi @katdom13 Can you please check the updates I made, thanks.

**Review by katdom13 — APPROVED** [review]
LGTM!

**Pull request merged**